### PR TITLE
feat: allow resources to be attached to workload collections

### PR DESCRIPTION
**Pull request by lander2k2**
Signed-off-by: Richard Lander <landerr@vmware.com>

### CODE REVIEW

File: docs/markers.md [hunk 1]
```diff
@@ -124,3 +124,13 @@ A second marker type `+operator-builder:collection:field` can be used with the
 same arguments as a Field Marker. These markers are used to define global Fields
 for your Collection and can be used in any of its associated components.
 
+Important: If you use a marker on a [collection
+resource](workload-collections.md#collection-resources) do *not* use the
```
Comment (scottd018) [inline docs/markers.md:128]: @lander2k2 should we add validation to the marker parsing to spit back a friendly error message? Or can we mutate the field into a collection field when we find a collection marker on a collection resource?  I think we may want to do the latter so we can avoid errors and bad UX when these markers are present.

**Review by scottd018 — APPROVED** [review]

File: docs/markers.md (hunk 1 above)
Comment (lander2k2) [reply docs/markers.md:128]: Yes, I agree.  I'll look into your second suggestion.  Failing that, I'll see if I can add validation.

Comment (lander2k2) [reply docs/markers.md]: @scottd018  Pushed a commit that treats all markers on collection resources as collection markers.

Comment (scottd018) [reply docs/markers.md]: @lander2k2 boom!  LGTM!

**Pull request merged**